examples/cms/src/FieldMapping.tsx

```diff
@@ -0,0 +1,244 @@
+                    } else {
+                        onFieldNameChange(field.id, value.trimStart())
+                    }
+                }}
+            />
+        </>
+    )
+}
+
+interface FieldMappingProps {
+    collection: ManagedCollection
+    existingFields: ManagedCollectionField[]
+    dataSource: DataSource
+    slugFieldId: string | null
+    onImport: typeof syncCollection
+}
+
+export function FieldMapping({ collection, existingFields, dataSource, slugFieldId, onImport }: FieldMappingProps) {
+    const originalFields = useMemo(() => computeFieldsFromDataSource(dataSource), [dataSource])
+    const [fields, setFields] = useState(mergeFieldsWithExistingFields(originalFields, existingFields))
+
+    const [disabledFieldIds, setDisabledFieldIds] = useState<Set<string>>(() => {
+        if (existingFields.length === 0) {
+            return new Set()
+        }
+
+        return new Set(
+            fields
+                .filter(field => !existingFields.find(existingField => existingField.id === field.id))
+                .map(field => field.id)
+        )
+    })
+
+    const possibleSlugFields = useMemo(() => {
+        return fields.filter(field => {
+            const isStringType = field.type === "string"
+            const isEnabled = !disabledFieldIds.has(field.id)
+
+            return isStringType && isEnabled
+        })
+    }, [fields, disabledFieldIds])
+
+    const [selectedSlugFieldId, setSelectedSlugFieldId] = useState<string | null>(
+        slugFieldId ?? possibleSlugFields[0]?.id ?? null
+    )
+
+    const [isSyncing, setIsSyncing] = useState(false)
+
+    const handleFieldNameChange = (fieldId: string, name: string) => {
+        setFields(prev => prev.map(field => (field.id === fieldId ? { ...field, name } : field)))
+    }
+
+    const handleFieldToggle = (fieldId: string) => {
+        setDisabledFieldIds(prev => {
+            const updatedDisabledFieldIds = new Set(prev)
+            const isEnabling = updatedDisabledFieldIds.has(fieldId)
+
+            if (isEnabling) {
+                updatedDisabledFieldIds.delete(fieldId)
+            } else {
+                updatedDisabledFieldIds.add(fieldId)
+            }
+
+            // Handle slug field updates
+            const field = fields.find(field => field.id === fieldId)
+            if (field?.type !== "string") {
+                return updatedDisabledFieldIds
+            }
+
+            if (isEnabling && (!slugFieldId || prev.has(slugFieldId))) {
+                // When enabling a string field and there is no valid slug field, make it the slug
+                setSelectedSlugFieldId(fieldId)
+            } else if (!isEnabling && fieldId === slugFieldId) {
+                // When disabling the current slug field, find next available one
+                const nextSlugField = possibleSlugFields.find(f => f.id !== fieldId)
+                setSelectedSlugFieldId(nextSlugField?.id ?? null)
+            }
```

Comment:
If we want to keep it working like this I think it would be much better to calculate the automatic slug field fallback int her render function instead of this callback. You can memoize that calculation if it's expensive

Reply:
I'd say it's fine to remove it now that we allow all source string fields to be slug candidates.